ros2component/test/test_cli_list.py

```diff
@@ -0,0 +1,145 @@
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2component-cli',
+                output='screen',
+                on_exit=[
+                    ExecuteProcess(
+                        cmd=['ros2', 'component', 'load', '/ComponentManager',
+                             'ros2component_test_fixtures', 'ros2component_test_fixtures::Talker'],
+                        additional_env={
+                            'RMW_IMPLEMENTATION': rmw_implementation,
+                            'PYTHONUNBUFFERED': '1'
+                        },
+                        name='ros2component-cli',
+                        output='screen'
+                    )
+                ]
+            )
+        ]
+    )
+    return LaunchDescription([
+        # Always restart daemon to isolate tests.
+        ExecuteProcess(
+            cmd=['ros2', 'daemon', 'stop'],
+            name='daemon-stop',
+            on_exit=[
+                ExecuteProcess(
+                    cmd=['ros2', 'daemon', 'start'],
+                    name='daemon-start',
+                    on_exit=[
+                        component_node,
+                        TimerAction(period=3.0, actions=[listener_command_action]),
+                        OpaqueFunction(function=lambda context: ready_fn())
+                    ],
+                    additional_env=additional_env
+                )
+            ]
+        ),
+    ]), locals()
+
+
+class TestROS2ComponentListCLI(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(
+        cls,
+        launch_service,
+        proc_info,
+        proc_output,
+        rmw_implementation
+    ):
+        @contextlib.contextmanager
+        def launch_component_command(self, arguments):
+            component_command_action = ExecuteProcess(
+                cmd=['ros2', 'component', *arguments],
+                additional_env={
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2component-cli',
+                output='screen'
+            )
+            with launch_testing.tools.launch_process(
+                launch_service, component_command_action, proc_info, proc_output,
+                output_filter=launch_testing_ros.tools.basic_output_filter(
+                    # ignore ros2cli daemon nodes
+                    filtered_patterns=['.*ros2cli.*'],
+                    filtered_rmw_implementation=rmw_implementation
+                )
+            ) as component_command:
+                yield component_command
+        cls.launch_component_command = launch_component_command
+
+    @launch_testing.markers.retry_on_failure(times=10)
+    def test_list_verb(self):
+        with self.launch_component_command(
+                arguments=['list']) as list_command:
+            assert list_command.wait_for_shutdown(timeout=30)
```

Comment:
@BMarchi wow, is this timeout arbitrarily long or did you really need 30 sec?

Reply:
I think I needed more retries rather than long timeout